apis/cluster/v1alpha1/types.go

```diff
@@ -0,0 +1,222 @@
+	// +optional
+	Password *string `json:"password,omitempty"`
+	// PasswordRef contains a reference to a kubernetes secret containing the Password
+	// +optional
+	PasswordRef *SecretReference `json:"passwordRef,omitempty"`
+	// Server requires Bearer authentication. This client will not attempt to use
+	// refresh tokens for an OAuth2 flow.
+	// +optional
+	BearerToken *string `json:"bearerToken,omitempty"`
+	// BearerTokenRef contains a reference to a kubernetes secret containing the BearerToken
+	// +optional
+	BearerTokenRef *SecretReference `json:"bearerTokenRef,omitempty"`
+	// TLSClientConfig contains settings to enable transport layer security
+	TLSClientConfig `json:"tlsClientConfig"`
+	// AWSAuthConfig contains IAM authentication configuration
+	// +optional
+	AWSAuthConfig *AWSAuthConfig `json:"awsAuthConfig,omitempty"`
+	// ExecProviderConfig contains configuration for an exec provider
+	// +optional
+	ExecProviderConfig *ExecProviderConfig `json:"execProviderConfig,omitempty"`
+}
+
+// SecretReference holds the reference to a Kubernetes secret
+type SecretReference struct {
+	// Name of the secret.
+	Name string `json:"name"`
+
+	// Namespace of the secret.
+	Namespace string `json:"namespace"`
+
+	// Key whose value will be used.
+	Key string `json:"key"`
+}
+
+// ClusterInfo holds information about cluster cache and state
+type ClusterInfo struct {
+	// ConnectionState contains information about the connection to the cluster
+	// +optional
+	ConnectionState *ConnectionState `json:"connectionState,omitempty"`
+	// ServerVersion contains information about the Kubernetes version of the cluster
+	// +optional
+	ServerVersion *string `json:"serverVersion,omitempty"`
+	// CacheInfo contains information about the cluster cache
+	// +optional
+	CacheInfo *ClusterCacheInfo `json:"cacheInfo,omitempty"`
+	// ApplicationsCount is the number of applications managed by Argo CD on the cluster
+	ApplicationsCount int64 `json:"applicationsCount"`
+}
+
+// TLSClientConfig contains settings to enable transport layer security
+type TLSClientConfig struct {
+	// Insecure specifies that the server should be accessed without verifying the TLS certificate. For testing only.
+	Insecure bool `json:"insecure"`
+	// ServerName is passed to the server for SNI and is used in the client to check server
+	// certificates against. If ServerName is empty, the hostname used to contact the
+	// server is used.
+	// +optional
+	ServerName *string `json:"serverName,omitempty"`
+	// CertData holds PEM-encoded bytes (typically read from a client certificate file).
+	// CertData takes precedence over CertFile
+	// +optional
+	CertData []byte `json:"certData,omitempty"`
+	// +optional
+	CertDataRef *SecretReference `json:"certDataRef,omitempty"`
+	// KeyData holds PEM-encoded bytes (typically read from a client certificate key file).
+	// KeyData takes precedence over KeyFile
+	// +optional
+	KeyData []byte `json:"keyData,omitempty"`
```

Comment:
To me it seems that they just used the default `TLSClientConfig` struct ([TLSClientConfig](https://pkg.go.dev/github.com/argoproj/argo-cd/v2@v2.0.1/pkg/apis/application/v1alpha1#TLSClientConfig))

It doesn't really make sense to have a Key field. Should we remove the field?

Reply:
We copy the argocd api